model/event/subscriptions.go

```diff
@@ -227,37 +367,18 @@ const (
 	SelectorInBuild      = "in-build"
 )
 
-var filterKeyForSelectorType = map[string]string{
-	SelectorObject:       filterObjectKey,
-	SelectorID:           filterIDKey,
-	SelectorProject:      filterProjectKey,
-	SelectorOwner:        filterOwnerKey,
-	SelectorRequester:    filterRequesterKey,
-	SelectorStatus:       filterStatusKey,
-	SelectorDisplayName:  filterDisplayNameKey,
-	SelectorBuildVariant: filterBuildVariantKey,
-	SelectorInVersion:    filterInVersionKey,
-	SelectorInBuild:      filterInBuildKey,
-}
-
-// FindSubscriptions finds all subscriptions of matching resourceType, and whose
+// FindSubscriptionsByAttributes finds all subscriptions of matching resourceType, and whose
 // filter and regex selectors match the attributes of the event.
-func FindSubscriptions(resourceType string, eventAttributes map[string][]string) ([]Subscription, error) {
-	if len(eventAttributes) == 0 {
+func FindSubscriptionsByAttributes(resourceType string, eventAttributes Attributes) ([]Subscription, error) {
+	if eventAttributes.isUnset() {
 		return nil, nil
 	}
 
+	// A subscription filter specifies the event attributes it should match.
+	// If a field is specified then it must match a corresponding attribute in the event.
```

Comment:
This wording seems kind of ambiguous about whether the "field" is a subscription filter or an attribute. Can this be rephrased as something like "If the subscription's filter specifies a field, then it must match... in the trigger's attributes"?

Reply:
I changed line 378 to 
> If the subscription's filter specifies a field then it must match one of the corresponding trigger attribute's values.